json_container/inc/leatherman/json_container/json_container.hpp

```diff
@@ -246,6 +243,25 @@ namespace leatherman { namespace json_container {
             return getValue<T>(*getValueInJson(keys, true, idx));
         }
 
+        /// Return the value of the specified entry of the root object,
+        /// or default_value if the entry doesn't exist.
+        /// Throw a data_type_error in case the type T doesn't match
+        /// the specified one.
+        template <typename T>
+        T getWithDefault(const JsonContainerKey& key, const T default_value) {
```

Comment:
This looks a lot like getValueInJson; just catching the data_key_error and returning a default_value.